### PR TITLE
🐛 Fix examples

### PR DESCRIPTION
**Pull request by je-cook**
## Linked Issues

Does this PR close or fix any Issues? Remember to create an Issue before starting work so that your fix / proposal can be addressed by the team.

e.g. Closes #{ID}

### Success/Failure
balance_of_plant/steady_state_example.py ✔️
base/components_tutorial.py ✔️
base/solver_tutorial.py ✔️
base/units.py ✔️
codes/run_plasmod_example.py ✔️
codes/solver_example.py ✔️
equilibria/eudemo_2017.py ✔️
equilibria/fem_fixed_boundary/equilibrium_example.py ✔️
equilibria/fem_fixed_boundary/fem_GS_Johner.py ✔️
equilibria/single_null.py ✔️
fuel_cycle/EUDEMO_fuelcycle.py ✔️
geometry/geometry_tutorial.py ✔️
geometry/optimisation_tutorial.py ✔️
geometry/plane_placement_tutorial.py ✔️
geometry/plotting_tutorial.py ✔️
materials/material_definitions.py ✔️
radiation_transport/heat_flux_calculation.py ✔️
radiation_transport/heat_flux_calculation_DN.py ✔️
structural/tour_eiffel_example.py ✔️
utilities/optimisation_example.py ✔️

### Crashes locally I think its just me
equilibria/fem_fixed_boundary/fem_magnetostatic_2D_single_coil.py ❌
mesh/mesh_tutorial.py ❌

### Crashes locally but runs in jupyter notebook form again just me?
magnetostatics/helmholtz_example.py ❌

### Needs redesign
design/EU-DEMO/EUDEMO_reactor.py ❌
design/design_plasma.py ❌
design/design_shapes.py ❌
equilibria/coilset_opt_problem_tutorial.py ❌

## Interface Changes

If you've had to update an interface or introduce a new interface as part of your change then let us know here.

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: examples/base/units.py [hunk 1]
```diff
@@ -62,20 +65,42 @@
...
+    A: Parameter[float]
+
+
+mypf = MyParameterFrame.from_dict({"A": {"value": 5, "unit": ""}})
+mydecpf = MyDecoratedFrame.from_dict({"A": {"value": 5, "unit": ""}})
+
+print(mypf)
+print(mydecpf)
+# Both frames equal
+assert all([dparam == param for dparam, param in zip(mydecpf, mypf)])  # noqa: S101
```
Comment (hsaunders1904) [inline examples/base/units.py:94]: I'm fairly sure you can do `mypf == mydecpf` here

**Review by hsaunders1904 — APPROVED** [review]
This all looks good to me. A couple of minor suggestions.

All the ones that are crashing for you, run without error for me.

I think we should delete all the "needs a redesign" examples, and open an issue (or issues) to re-implement them. If they don't work, they probably shouldn't be in develop

File: examples/base/units.py (hunk 1 above)
Comment (je-cook) [reply examples/base/units.py:94]: image

Apparently not...

File: bluemira/codes/plasmod/api/_outputs.py [hunk 2]
```diff
@@ -153,6 +153,8 @@ def read_plasmod_output(io_stream: TextIO) -> Dict[str, Union[np.ndarray, float]
         output_key, *output_value = row[0].split()
         if len(output_value) > 1:
             output[output_key] = np.array(output_value, dtype=float)
+        elif len(output_value) == 0:
+            output[output_key] = None
```
[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/codes/plasmod/api/_outputs.py:157]: It probably should be an empty numpy array. I managed to run a plasmod and then output printed the key but had no data associated with it

File: examples/base/units.py (hunk 1 above)
Comment (hsaunders1904) [reply examples/base/units.py:94]: Oh dear, that's not good

**Comment by je-cook** [comment]
issues made

File: examples/base/units.py (hunk 1 above)
Comment (je-cook) [reply examples/base/units.py:94]: I've made an issue for now see #1539

**Pull request merged**